torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -510,6 +517,8 @@ struct NCCLTraceBuffer {
       dict.insert(input_sizes_s, read_sizes(e.input_dims_));
       dict.insert(output_sizes_s, read_sizes(e.output_dims_));
       dict.insert(state_s, e.state_);
+      dict.insert(retired_s, e.retired_);
```

Comment:
is adding 'retired' to the log really needed?  ideally we'd just avoid putting 'completed' in the log as the state for these cases, and then not need this.  or what's the full set of combinations that can happen?

Reply:
I think when everything is working inside the nccl bindings it doesn't matter too much, but knowing which entries are still in the workMetaList_ and were queried, vs which were already removed might be a useful signal on its own for understanding the state when a timeout happened.